src/components/Provider.tsx

```diff
@@ -34,12 +34,36 @@ interface IStyedThemeColorOptions {
   themeColors: ThemeColors;
 }
 
-const SName = styled.div<IStyedThemeColorOptions>`
+const SName = styled.button<IStyedThemeColorOptions>`
   width: 100%;
   font-size: 24px;
   font-weight: 700;
+  overflow: hidden;
+  background-color: transparent;
+  border: none;
+  outline: none;
   margin-top: 0.5em;
   color: ${({ themeColors }) => themeColors.main};
+
+  @media (hover: hover) {
+    &:hover, &:focus {
+      &::before {
+        background-color: ${({themeColors}) => themeColors.hover};
+      }
+    }
+  }
+
+  &::before {
```

Comment:
This is just a trick when we need to make clickable area larger than the element. So we make this pseudo element stretch to the full width of provider card.